core/src/main/scala/org/apache/spark/shuffle/ShuffleBlockResolver.scala

```diff
@@ -39,3 +36,5 @@ trait ShuffleBlockResolver {
 
   def stop(): Unit
 }
+
+private[spark] case class ShuffleIdAndAttempt(shuffleId: Int, stageAttemptId: Int)
```

Comment:
this could be `private[shuffle]` except that its needed in [`StoragePerfTester`](https://github.com/squito/spark/blob/SPARK_8029_shuffleoutput_per_attempt/tools/src/main/scala/org/apache/spark/tools/StoragePerfTester.scala#L64) -- I'm not familar enough with it to suggest a potential refactoring.  in any case `private[spark]` isn't too bad.